pedalboard/process.h

```diff
@@ -57,40 +57,51 @@ processSingle(const py::array_t<SampleType, py::array::c_style> inputArray,
                              reset);
 }
 
-/**
- * Process a given audio buffer through a list of
- * Pedalboard plugins at a given sample rate.
- * Only supports float processing, not double, at the moment.
- */
-template <>
-py::array_t<float>
-process<float>(const py::array_t<float, py::array::c_style> inputArray,
-               double sampleRate, const std::vector<Plugin *> &plugins,
-               unsigned int bufferSize, bool reset) {
+template <typename T>
+ChannelLayout
+detectChannelLayout(const py::array_t<T, py::array::c_style> inputArray) {
+  py::buffer_info inputInfo = inputArray.request();
+
+  if (inputInfo.ndim == 1) {
+    return ChannelLayout::Interleaved;
+  } else if (inputInfo.ndim == 2) {
```

Comment:
Maybe add some log here about inferring a shape. May help in future debugging. I would consider having a default layout and let the user override the layout instead.